### PR TITLE
[#46439] Bahn Plugin für OpenProject 12.5.x anpassen

### PR DESCRIPTION
**Pull request by ba1ash**
https://community.openproject.org/work_packages/46439

### CODE REVIEW

File: frontend/src/app/shared/components/attachments/attachment-list/attachment-list-item.component.ts [hunk 1]
```diff
@@ -167,6 +167,13 @@ export class OpAttachmentListItemComponent extends UntilDestroyedMixin implement
     return el;
   }
 
+  public get linkHref():string {
+    if (this.attachment._links.originOpen !== undefined) {
+      return this.attachment._links.originOpen.href;
+    }
+    return this.attachment._links.staticDownloadLocation.href;
+  }
+
```
Comment (Kharonus) [inline frontend/src/app/shared/components/attachments/attachment-list/attachment-list-item.component.ts]: I'd probably remove this extra getter and instead just use in the template:

```html
[href]="attachment._links.originOpen?.href || attachment._links.staticDownloadLocation.href"
```

File: frontend/src/app/shared/components/attachments/attachment-list/attachment-list-item.component.html [hunk 2]
```diff
@@ -28,10 +28,18 @@
...
+    <a
+      *ngIf="!!this.attachment._links.originOpen"
+      class="spot-link"
+      [href]="this.attachment._links.staticDownloadLocation.href"
+      target="_blank"
+    >
+      <span class="spot-icon spot-icon_download-arrow"></span>
+    </a>
     <button
+      *ngIf="!!this.attachment._links.delete"
```
Comment (Kharonus) [inline frontend/src/app/shared/components/attachments/attachment-list/attachment-list-item.component.html]: you do not need to `this.` in the template

**Review by Kharonus — CHANGES_REQUESTED** [review]
I'd like to reduce a bit the code used in here.

File: frontend/src/app/shared/components/attachments/attachment-list/attachment-list-item.component.ts (hunk 1 above)
Comment (ba1ash) [reply frontend/src/app/shared/components/attachments/attachment-list/attachment-list-item.component.ts]: Cool!

**Review by Kharonus — APPROVED** [review]
LGTM

**Pull request merged**